### DISCUSSION TITLE
olympiad-computer-engineering

### DISCUSSION THREAD

**Discussion by giscus** [amir-pourmand/hugo-papermod-farsi-template#6] [Announcements]
# https://aprd.ir/olympiad-computer-engineering/

تجربیات من در مورد المپیاد مهندسی کامپیوتر در ۲ سال ...

https://aprd.ir/olympiad-computer-engineering/

**Comment by pourmand1376** [comment]
**نویسنده کامنت:  [آیناز - ویرگول](https://virgool.io/@m_28262561)
**متن کامنت:**
سلام آقای پورمند  
از اطلاعات خوبتون خیلی ممنونم  
یک سوال داشتم ؛ اگه موقع امتحان هنوز ۷۰ واحد رو نگذرونده باشیم( یعنی تو اون ترم در حال پاس کردن اون ۷۰ واحد باشیم، مثلا من این ترم رو که بگذرونم میشه ۶۸ واحد ولی موقع برگزاری امتحان تعداد واحد های تو سبد درسیم بیشتر از ۷۰ واحد میشه) ولی دانشجوی سال سوم باشیم، میتونیم تو المپیاد شرکت کنیم؟  
این مشکل رو به این خاطر دارم که تو ترم اول ۱۲ واحد پیش خوردم که متاسفانه جزو واحد های گذرونده حساب نمیشن

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
سلام آیناز
تا اونجا که یادمه، برای من اینطوری بود که صرفا چون درس‌های المپیادی رو پاس کرده بودم می‌تونستم در آزمون شرکت کنم. زمان ما خبری از قانون ۷۰ واحد نبود.

به نظرم این قانون برای اینه که راحت‌تر بتونند دانشجو رو انتخاب کنند. نهایتا دانشگاه باید پنج نفر به منطقه معرفی کنه و منطقه هم اصلا براش مهم نیست که اینایی که انتخاب شدند، درس‌هاشون رو پاس کردند یا نه و این که چند واحد پاس کردند.

به نظرم با استاد مسئولش صحبت کن و شرایطت رو بهش توضیح بده. به احتمال زیاد می‌تونه این موضوع رو برات حل کنه. بعید می‌دونم به خاطر دو واحد کمتر مانع از شرکت در المپیاد بشن. مخصوصا که مشخصه انگیزه کافی رو داری :)

امیدوارم موفق باشی.

**Comment by pourmand1376** [comment]
**نویسنده کامنت:  [ali mohammadi - ویرگول](https://virgool.io/@ali.mohammadi.20779)
**متن کامنت:**
سلام . خسته نباشید . ممنون از توضیحاتتون . یک سوال داشتم . اینکه برای هر مبحث تایین شده تا کجا باید بخونیم ؟؟؟؟؟ تا جایی که دانشگاه بهمون در باره اون مبحث درس داده یا کل کتابی که مربوط به اون مبحث میشود ؟؟؟؟  
  
لطفا به سوالم جواب بدید 🙏🙏🙏

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
سلام علی
به سایت وزارت علوم برو تو آدرس زیر و سرفصل مصوب فعال وزارت علوم رو دانلود کن و ببین برای هر درس چه مباحثی مطرح شده. متر و معیار همونه. در واقع سال ما سرفصل مصوب امیرکبیر بود که تو امیرکبیر تصویب شده بود و تدریس می‌شد.

اینها ممکنه هر سال فرق کنه.
https://prog.msrt.ir/fa/grid/113/%D9%84%DB%8C%D8%B3%D8%AA-%DA%A9%D8%A7%D9%85%D9%84-%D8%B1%D8%B4%D8%AA%D9%87-%D9%87%D8%A7?GridSearch%5Bsearch%5D=%D9%85%D9%87%D9%86%D8%AF%D8%B3%DB%8C+%DA%A9%D8%A7%D9%85%D9%BE%DB%8C%D9%88%D8%AA%D8%B1&GridSearch%5BpageSize%5D=10&page=1&per-page=10

**Comment by pourmand1376** [comment]
**نویسنده کامنت: [mehrnaz jiryaee - ویرگول](https://virgool.io/@2000mehrnaz)
**متن کامنت:**
سلام و وقت بخیر  
یک جا گفتید که سوالات المپیاد بیشترشون اثباتیه، اگر اثبات و استدلالمون ضعیف باشه چطوری تقویتش کنیم؟  
(برای کسی که المپیاد براش جدیه و میخواد نتیجه عالی کسب کنه)

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
سلام.
اولین چیزی که به ذهنم می‌رسه کتاب دکتر قدسیه که خیلی می‌تونه کمک کننده باشه. فیلم های اساتید خوب (مثل پروفسور نوابی، دکتر گودرزی و ...) رو هم ببینید که درک عمیق پیدا کنید از مطالب.

اگر وقت کردید تمرین‌های کتاب رفرنس و خود کتابهای رفرنس رو هم بخونید. البته تو این قسمت باید بهینه مطالعه کنید و همه‌اش رو نخونید وگرنه وقت نمی‌کنید تموم کنید.

من خودم اگر بخوام بخونم قبل از همه اینها مجموعه درس‌های «مهارت یادگیری» تو متمم رو مطالعه ‌‌‌‌می‌کنم چون بهینه یادگرفتن خیلی مهمه. قرار نیست روزی ۱۰ ساعت وقت بذارم ولی می‌خوام همون مقداری که وقت می‌ذارم، مفید باشه.

**Reply by pourmand1376** [reply]
**نویسنده کامنت: [mehrnaz jiryaee - ویرگول](https://virgool.io/@2000mehrnaz)
**متن کامنت:**
ممنون که انقدر زود جواب میدید.  
- منظور از کتاب دکتر قدسی هر دو کتابیه که معرفی کردید یا فقط یکیش مدنظرتونه ؟  
و اینکه منظورتون از فیلم های دکتر گودرزی مال مکتب خونه هست ؟  
لینکش : [https://maktabkhooneh.org/course/%D9%85%D8%B9%D9%85%D8%A7%D8%B1%DB%8C-%DA%A9%D8%A7%D9%85%D9%BE%DB%8C%D9%88%D8%AA%D8%B1-mk51/](https://maktabkhooneh.org/course/%D9%85%D8%B9%D9%85%D8%A7%D8%B1%DB%8C-%DA%A9%D8%A7%D9%85%D9%BE%DB%8C%D9%88%D8%AA%D8%B1-mk51/)

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
کتاب داده ساختار کتاب ساده ای هست. منظورم کتاب ۶۰۰ مسئله است.

بله. فیلم های دکتر گودرزی از مکتب خونه.

**Comment by pourmand1376** [comment]
**نویسنده کامنت: [mehrnaz jiryaee - ویرگول](https://virgool.io/@2000mehrnaz)
**متن کامنت:**
سلام ممنون از اطلاعات خوبتون  
چنتا سوال داشتم:  
برای اجازه شرکت توی المپیاد غیر متمرکز حتما باید معدل بالا باشیم؟  
برای آمادگی برای المپیاد بنظرتون یکسال تایم خوبه؟

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
سلام.
۱- مرحله اولیه اش دست مسئول دانشگاس. ممکنه با معدل تصمیم گیری کنن ممکنه با آزمون. هر دانشگاهی فرق داره. مراحل بعدی هم که با آزمونه.
۲- اره. حتما کافیه :)

**Reply by pourmand1376** [reply]
**نویسنده کامنت: [mehrnaz jiryaee - ویرگول](https://virgool.io/@2000mehrnaz)
**متن کامنت:**
یک سوال دیگه،  
علاوه بر مزایایی که گفتید برای اپلای به دانشگاه های معتبر خارج از ایران بدرد میخوره؟  
و اینکه ایا این المپیاد های دانشجویی مرحله جهانی هم دارند؟

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
راجع به سوال اول. راستش میگن به درد میخوره ولی ما که ندیدیم!
و نه مرحله جهانی ندارند.

**Comment by pourmand1376** [comment]
**نویسنده کامنت: [علی میکائیلی - ویرگول](https://virgool.io/@alim98barnet)
**متن کامنت:**
سلام خسته نباشید ببخشید دانشجویانی که سال سوم صرفا برای المپیاد شرکت میکنند تو کنکور ارشد، از رشته شناور هم اگه رتبه آوردن دعوت میشن؟  
مثلا اگر کسی هم کنکور کامپیوتر بده هم آی تی اگه از آی تی رتبه بیاره دعوت میشه؟

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
سلام. مطمئن نیستم ولی بعید میدونم رتبه رشته شناور حساب بشه. بهتره به [olympiad@sanjesh.org](mailto:olympiad@sanjesh.org) ایمیل بدید و بپرسید.

**Comment by pourmand1376** [comment]
**نویسنده کامنت: [m\_25144478 - ویرگول](https://virgool.io/@m_25144478)
**متن کامنت:**
سلام ممنونم از اطلاعات مفیدی که دادین  
یک سوال داشتم اینکه نوشته بودید سال سوم و چهارم امکان شرکت در امپیاد هست و یعنی میشه دو سال پشت سر هم شرکت کرد؟

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
بله. چرا که نه. منم همین جوری ۲ بار شرکت کردم.

**Comment by pourmand1376** [comment]
**نویسنده کامنت: [ابوالفضل - ویرگول](https://virgool.io/@m_73484264)
**متن کامنت:**
امیر جان سلام خیلی ممنون بابت به اشتراک گذاشتن تجربت میخواستم بدونم بچه های علوم کامپ که وسط کارشناسی هستن هم میتونن توی المپیاد شرکت کنن؟ اگر بله درصورت اینکه رتبه اوردن برای مقطع بعدی میتونن هر دانشگاهی رو انتخاب کنند درسته؟ مرسی از لطفت

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
سلام اقا ابوالفضل. خواهش میکنم.
بله فکر میکنم بتونید از علوم کامپیوتر هم شرکت کنید ولی مطمئن نیستم. بهتره از مسئول استعداد درخشان دانشکده تون بپرسید. اونها بهتر میدونن.
بابت سوال دومتون هم بله اگر رتبه یک تا پنج بیارید هر جایی میتونید برید. البته امسال ظاهرا عوض شده بود یک تا پانزده میتونستن هر جایی برن. سازمان سنجش یخورده تکلیفش با خودش مشخص نیست.

**Comment by pourmand1376** [comment]
**نویسنده کامنت: [Adelzde - ویرگول](https://virgool.io/@zhraadly464)
**متن کامنت:**
سلام ممنون بابت اطلاعاتی ک در اختیارمون گذاشتین .  
میشه بگین فیلم های پروفسور نوابی رو از کجا میشه تهیه کرد؟ من سرچ کردم فقط زبانگ انگلیسیشو پیدا کردم شما مدنظرتون زبان انگلیسیه یا فارسیشم موجوده؟

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
سلام. خسته نباشید.

به نظرم یه نگاه دوباره ای تو مکتب خونه بکنید. دو دسته فیلم بود. یکی انگلیسی بود که خیلی پسندم نشد چون با بچه ها حالت مشارکتی نداشت. اون یکی سرکلاس دانشگاه تهران ضبط شده بود که من این مدنظرمه. حالا بازم اگر پیدا نشد دوباره پیام بدید ببینیم چی شده.

**Reply by pourmand1376** [reply]
**نویسنده کامنت: [Adelzde - ویرگول](https://virgool.io/@zhraadly464)
**متن کامنت:**
ممنون از پاسختون ?

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Comment by pourmand1376** [comment]
**نویسنده کامنت: [Fetching Title#icue](https://virgool.io/@m_40324565)
**متن کامنت:**
سلام اقا امیر  
دمت گرم مطالب خیلی خوبی بود  
یه سوال داشتم، روش شرکت در المپیاد با کنکور، منظور اینه خود امتحان کنکور رو باید بدیم جزو ۱۵ نفر برتر بشیم یا از کنکور امتحانش جداست ؟

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
سلام خواهش میکنم.

اره همینه. همون کنکور رو که میدین خودشون یه سری ها رو مشخص میکنن برای المپیاد. راه اطلاع رسانیشون هم پیامکی هست که سازمان سنجش بعد کنکور براتون میفرسته. فقط لزوما این نیست که نفر ۱ تا ۱۵ باشه. ممکنه یکی با رتبه ۵۰ هم دعوت بشه. فرمولشون رو خیلی نفهمیدم چطوریه.

**Comment by pourmand1376** [comment]
**نویسنده کامنت: [علی میکائیلی - ویرگول](https://virgool.io/@alim98barnet)
**متن کامنت:**
سلام اقای پورمند خدا قوت خیلی ممنون از مطالبتون خیلی به من انگیزه میده حرفای شما امیدوارم همیشه موفق باشید

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Comment by pourmand1376** [comment]
**نویسنده کامنت: [mohaddeseh - ویرگول](https://virgool.io/@m._.m._.t._.k)
**متن کامنت:**
سلام اقاي پورمند وقت بخير  
من ترم4كارشناسي ازاد تبريز كامپيوترم ميخواستم براي مرحله اول المپياد اماده بشم بنظرتون خيلي ديره منصرف بشم ؟ ويا چون دانشگاه ازادي هستم حق شركت در اين ازمون ندارم؟  
سوال بعديم اينكه در متمركز به دو صورت شركت ميكنن يكي سال سومي ها هستن يكي هم سال چهارمي ها من ميخواستم كه در ازمون ارشد شركت كنم بايد سال چهارمي باشم يا كدوم يك از اين دو روش هست؟  
هزينه خريد كتاب ها وفيلم هايي كه گفتيد تاحدودي چقدر ميشه و چطوري برنامه ريزي كنم كه بتونم برسم هدف من اينه مرحله اولش شركت كنم  
من پكيج كنكوري براي كنكور ارشد خريدم بعضي درس هاشو وچون مجازي شده كيفيت تدريس اساتيد پايينه از طرفي دانشگاهم سطحش خوب نيست اولويت اول من اين بود اول اين درس هايي كه از اين پكيج كنكوري خريدم نگاه كنم بعد برم سراغ منابع و رفرنس اصلي كه شما اينجا گفتيد بنظرتون اين كارم درسته ؟ يا ميتونم خودمو براي مرحله اول برسونم ؟  
ممنونم از راهنماييتون

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
سلام خسته نباشید. امیدوارم حالتون خوب باشه.
در مورد سوال اولتون راستش نمیدونم باید از مسئول استعداد درخشان دانشکده یا دانشگاه بپرسید.
در مورد سوال دومتون، شما میتونید به هر دو روش شرکت کنید مشکلی نداره. مثلا من خودم هم از طریق منطقه شرکت کردم و هم از طریق کنکور!
هزینه خرید فیلم و کتاب خیلی بستگی به تورم داره. از ۲ سال پیش که من کتاب خریدم کتابها چندین برابر شده و تخمین من برای شما اعتباری نداره. ولی به عنوان قاعده کلی هزینه ای که برای کنکور ارشد براتون درمیاد خیلی خیلی کمتر از هزینه کنکور سراسریه. هم کتاب هاش تا حدی ارزون ترن و هم فیلم های آموزشی.

به نظرم توقعتون از دانشگاه و مطالبی که اونجا ارائه میشه رو بیارید پایین تر. فیلم های اموزشی کنکوری بعضی موسسات خیلی خوبن. واقعا مفیدن. نمیدونم جایی اشاره کردم یا نه. به نظرم فیلم های خود اقای یوسفی در کنکور کامپیوتر به نظرم بی نظیرن.

کتاب های رفرنس رو بهتره برای کنکور صرفا به جایی که مشکل جدی دارید و با کتاب های کنکوری نمیفهمید محدود کنید . واقعا وقت این کارها نیست. با همون فیلم و کتاب کنکوری پیش برید و یادتون نره ... خیلی خیلی زیاد تست بزنید ... تست زدن برای مروره.... تست زدن اصلا برای یادگیریه....

**Reply by pourmand1376** [reply]
**نویسنده کامنت: [mohaddeseh - ویرگول](https://virgool.io/@m._.m._.t._.k)
**متن کامنت:**
سلام ممنون از پاسخ و راهنماییتون . اگر برای المپیاد در ازمون ارشد شرکت کنم باید تمامی دروس کنکور ارشد بخونم یا همین چهار درسی که گفتید رو ؟ و ایا بنظرتون در عرض یکسال میشه این چهار درسی که گفتید رو مسلط شد و تمامشون کرد ؟

----
این کامنت از سایت ویرگول که قبلا استفاده می‌کردم استخراج شده است.

**Reply by pourmand1376** [reply]
برای المپیاد از طریق کنکور باید رتبه خوب بیارید و رتبه قاعدتا به بقیه درس‌ها هم وابسته اس. (البته تقریبا یه کار اضافی حساب میشه المپیاد دیگه چون با رتبه خوب میشه هر دانشگاهی رفت! و خب تنها فایده‌اش امتیاز بنیاد ملی نخبگانش میشه)

یک سال؟ ۴ درس؟ تو یک سال خیلی بیشتر از این ۴ تا رو هم میشه مطالعه کرد. با برنامه ریزی درست حسابی میشه اکثر درس‌های کنکور رو از تقریبا صفر تو یکسال خوند.

**Comment by codesamira** [comment]
سلام وقتتون بخیر منابع اصلی ای ک معرفی کردید (مایکل سیپسر و پیتر لینز)پیشنهاد میدید که کتاب زبان اصلی مطالعه بشه یا ترجمه هاشون و اگر مورد دوم پاسختون بود چه انتشاراتی ترجمه های خوبی از این کتاب ها دارند ؟

**Reply by pourmand1376** [reply]
سلام. وقت بخیر. 

قطعا خواندن کتاب زبان اصلی حس و حال متفاوتی داره. اما فکر می‌کنم بیشتر به سلیقه برمی‌گرده و این که چقدر راحت باشید.

من به شخصه با مطالعه زبان اصلی مفاهیم خیلی انتزاعی راحت نیستم و ترجیح میدم همون فارسی رو بخونم. 

برای ترجمه هم، من ترجمه خاصی مد نظرم نیست و قبلاً همون کتابی که از کتابخانه پیدا کردم رو خوندم. الان به طور خاص برای پیتر لینز جستجو کردم و دیدم انتشارات نص ترجمه داره. 

من قبلاً چند کتاب از انتشارات نص گرفتم و انصافا ترجمه‌هاشون دقیق هست. 

اما برای مایکل سیپسر ترجمه خاصی مد نظرم نیست چون من اصلا کتابش رو نخوندم. 

تو آزمایشگاه دیدم که بعضا به بچه‌های شریف همین کتاب مایکل سیپسر رو درس میدن. شاید اگر تو مکتب خونه نگاه کنید بتونید کورس آنلاین یا ضبط شده هم پیدا کنید. 

امیدوارم پاسخ سوالتون رو داده باشم.

با مهر 
امیر

**Comment by MRK118** [comment]
سلام جناب پورمند . وقتتون بخیر .
آقای پورمند ، از منابعی که فرمودید ، دو کتاب داده ساختار و الگوریتم و ۶۰۰ مسئله آقای قدسی ، سیستم عامل آقای حقیقت ، شبکه کوروس و راس و فیلم های آقای ماش همدانی و پروفسور نوابی پیدا کردم ! مشکل اصلی من با کتاب ها است ! نمی‌دونم باید واو به واو کتاب بخونم یا مباحث انتخابی بخونم یا ...؟ اصلا نمی‌دونم مثل کنکور سراسری بخونم مثل زیست (چون رشتم تجربی بوده !) یا ...؟ چطور بهینه بخونم ؟ لطفا راهنماییم بفرمایید. همچنین منابع دیگری هم هست از کتب رفرنس که بخونم ولی نمی‌دونم آیا همین منابع درصورت خوب خونده شدن ، می‌تونه منو به رتبه برتر المپیاد برسونه یا نه ؟ واقعیتش تا الان نه منابع ریاضی و محاسباتی بصورت کنکوری خوندم نه اصلا برای المپیاد خوندم اگر شیوه مطالعه و حدودا ساعت مطالعاتی این المپیاد بفرمایید واقعا ممنون میشم ! حقیقتا سرم خیلی شلوغ و بشدت نیاز به بهینه مطالعه کردن دارم ! متشکرم 🙏

**Comment by MRK118** [comment]
ضمن اینکه سراغ هر کتاب فروشی رفتم ، سخت میتونستم این کتب گیر بیارم ، میگفتند این منابع قدیمی اند و زیاد دیگه نیستند ! همچنین برخی از شبه کد ها یا ادبیات و موضوعاتش چون یا باهاشون برخورد نداشتم یا متوجه نمیشم ، سخت اند برام خوندنشون ! نگران هم هستم که برای فهم سراغ منبع دیگری هم برم و مطلب یا نکته ای از این کتب از دست بدهم مخصوصا کتاب دکتر قدسی  🤦😫😂! یه راهنمایی بفرمایید خواهشاً جناب پورمند . ممنونم آقای مهندس

**Reply by pourmand1376** [reply]
سلام. 

راستش هدفم از نوشتن این مطالب این نبوده که بگم دقیقا این کتاب‌ها رو بخونید و رتبه بیارید (هر چند خودم با خواندن همین کتاب‌ها رتبه آوردم). 

صادقانه بخوام بگم من هیچ کدوم از این کتاب‌ها رو کامل نخوندم. یادگیری‌ام هم بیشتر سوال محور بوده. یعنی یک سوالی رو می گذاشتم جلو و سعی می‌کردم حل کنم. 

حالا در حین‌اش از منابع مختلف استفاده می‌کردم. اگر فلان مطلب از فلان کتاب بد بود و نمی‌فهمیدم، از یه کتاب دیگه می‌خوندم. اگر نمیشد از یوتیوب یاد می‌گرفتم. اگر نه کتاب فلانی.

تو این مسیر کتاب‌هایی مثل ۶۰۰ مسئله دکتر قدسی یا تمرین‌های CLRS و هر کتاب مرجع دیگه ای رو می‌تونی در نظر بگیری و سعی کنی اون‌ها رو حل کنی. 

مثلاً اگر می‌خوای نظریه بخونی به جای این که بشینی هی حفظ کنی و آخرش هم نفهمی، برو سوالات نظریه ته کتاب پیتر لینز رو حل کن. یا حتی اگر دیدی برات سخته از سوالات کنکوری نظریه شروع کن. 

سیستم عامل و معماری و الگوریتم هم همین‌طور. برو از یک کتابی یا مرجعی که حال می‌کنی یک سری سوال جون‌دار پیدا کن و یا علی.

نمیشه گفت این روش بهترین یا حتی موثرترین روش هست. فقط روشی هست که من در پیش گرفتم.

**Comment by MRK118** [comment]
واقعا ممنونم مهندس جان ! متشکرم 🙏

در تاریخ دوشنبه ۶ نوامبر ۲۰۲۳،‏ ۱۸:۳۴ Amir Pourmand <
***@***.***> نوشت:
